### PR TITLE
Semantic changes and thoughts on requirements

### LINKED ISSUE #3: Simplify boilerplate

### PR DESCRIPTION
**Pull request by khumphrey**
closes #3

@Kmacpher @cassiozen I updated the server to match what they already know (I think we can save the new things for the boilerplate :D ). I also updated the README to say that React is the only thing required, and I noted that this would mean they need to do a LOT of work to be ready in senior phase. Using redux without react-redux isn't something I think we should really teach them I don't think because people don't use it like that (is my understanding)

### CODE REVIEW

**Comment by khumphrey** [comment]
@omriBernstein mentioning you as well because you are often good for a review :D

File: README.md [hunk 1]
```diff
@@ -20,23 +19,26 @@ You are the CTO of the Margaret Hamilton Interplanetary Academy of JavaScript. C
 
 ### The tools
 
-Use at least sequelize, express, react, and redux when creating this app. You can incorporate any additional libraries or tools you wish.
+Use at least Sequelize, Express and React when creating this app. This app is small, so just using React is reasonable, but note that without practicing with more frontend libraries you will have a steep learning curve in senior phase. If you are going forward with ***just React*** you need to **delete all references to React-Redux, Redux and React-Router** (you either do the whole project with redux or none at all).
```
Comment (omriBernstein) [inline README.md]: Haha first thing is that I can literally hear you saying these words, especially with where ***emphasis*** is added.

But more relevantly: why does redux have to be an all-or-nothing choice? Is this just to guide them effectively—so they don't run into bugs with redux if they're intending to exclude it?

File: README.md [hunk 2]
```diff
@@ -20,23 +19,26 @@ You are the CTO of the Margaret Hamilton Interplanetary Academy of JavaScript. C
...
   * must be assigned to a campus
 
 - Campuses
-  * have info such as a name, imageUrl, and description
-  * name should not be allowed to be empty or null
-  * imageUrl should have a default value
-  * description should allow an extremely large description
+  * have profile info including:
+    * name - not empty or null
+    * imageUrl - defaul value
```
Comment (omriBernstein) [inline README.md]: Just like the British on May 10, 1773—missing `t`. Also I think we should say something like "should have a default value" for clarity

File: README.md [hunk 3]
```diff
@@ -20,23 +19,26 @@ You are the CTO of the Margaret Hamilton Interplanetary Academy of JavaScript. C
...
+    * description - extremely large text
```
Comment (omriBernstein) [inline README.md]: # OK

File: README.md (hunk 1 above)
Comment (khumphrey) [reply README.md]: If they are half managing application state with redux and then freak out and start doing everything locally - I think it makes more sense to just make it all with just React and pull everything else out. But I guess they don't **need** to delete everything. I am concerned they are going to confuse themselves more trying to do half this and half that

Comment (omriBernstein) [reply README.md]: Right, makes sense—I agree. If so, I'd recommend changing the wording so it comes across as a recommendation (however strong) as opposed to a constraint or a requirement of the project.

**Comment by omriBernstein** [comment]
I'm having altogether too much fun with this code view. Anyways, everything looks generally great, I have no additional comments on top of the inline comments I already put in!

**Comment by Kmacpher** [comment]
I'm still not sure I agree that letting them opt for just React (with no router or redux) is a good choice. Did we decide that for sure at some point? First, I think it's possible that it's more confusing to go back to just plain react after finishing up juke and stackchat, but they might be tempted to do so since it seems easier. Second, even if they do a great job with only react, they might just not be ready for senior phase projects, even if you stress that they need to study it.  

I'm very willing to be convinced of the merit of allowing react only, but not quite there yet.

**Comment by omriBernstein** [comment]
Ideologically I'm pretty close to where you're at right now @Kmacpher. Though speaking from the convinced-but-just-barely camp, I feel good about redux being optional mainly because of the timing.

When we open the senior enrichment for them, I think they haven't even been introduced to redux yet (assuming the schedule is still similar to when I last ran junior phase), so it's always tough to message: "hey get started...but don't all the frontend stuff yet". And furthermore, even by the time the senior enrichment project is due, they haven't been working with redux for that long. Some people take a little while to internalize and digest—in fact, I recall students saying they feel like they "actually get redux" after _Grace Shopper_ in senior phase.

As for whether it's simpler or not without redux, anecdotally I remember some students reporting they found it easier. That they would trip themselves up trying to remember all the right redux/react-redux syntax/usage and found it simpler to just use plain old react state.

Oh and I don't recall if or when we came to a decision that redux could be optional (I _think_ we did decide on this at some point, but I'm not sure).

...and I do agree that if we were truly testing them for preparedness for senior phase, redux _should_ be part of that. I just think if so, we would need to give them more time to digest it.

**Comment by Kmacpher** [comment]
Hm. Reasonable reasoning. I suppose we can try it and then see how well those students do with react after Grace Shopper. It's true that they really only get a few days after learning redux to implement it. 

The reasoning would apply well to not having redux, but also no react-router?

**Comment by omriBernstein** [comment]
Oh! I didn't read closely enough it seems. Yeah I'm didn't realize react router would be optional...

**Comment by khumphrey** [comment]
@cassiozen @danceoval and I have definitely agreed at one point or another on the fact that this is a VERY small application that if we were building as engineers, might as well be *only* React because why add the bulk when there is no need. And if that is what you would do IRL then....

We obv focus on education, and also they barely know redux. I could be convinced that they are required to use React Router, but we did NOT have that as a requirement before this time

**Comment by omriBernstein** [comment]
Woops right that makes sense, I forgot we _already_ didn't mandate using react router

**Comment by Kmacpher** [comment]
I definitely have always assumed we had! And I think my students have too..

**Comment by khumphrey** [comment]
That's great though @Kmacpher :D 

Just as an fyi - @cassiozen and I last phase allowed them to just use React and we had maybe 5 students start with that (out of 43). It was really beneficial for those people honestly. And also, the way we framed it - that using everything will be best and prepare you most - encouraged everyone else to use everything. This is just one sample size, but I think Cassio has done this with a previous group as well

**Comment by khumphrey** [comment]
@Kmacpher @tmkelly28 @omriBernstein @cassiozen @johnptmcdonald @connieelee Let me know any final thoughts on this? I'd love to merge some bits of it at least before they jump into it

File: app/main.jsx [hunk 4]
```diff
@@ -4,7 +4,7 @@ import {render} from 'react-dom'
 import { Provider } from 'react-redux'
 
 import store from './store'
-import Root from './components/Root'
+import Root from './components/Root.jsx'
```
Comment (connieelee) [inline app/main.jsx]: is there a reason for adding extensions to imports?

**Comment by omriBernstein** [comment]
No new comments, merge away as far as I'm concerned

File: app/main.jsx (hunk 4 above)
Comment (khumphrey) [reply app/main.jsx]: My understanding is that not including the extension only works with `.js` and I switched to if files had `jsx` in them they had that extension

Comment (collin) [reply app/main.jsx]: We would have to add `.jsx` to the webpack resolver to get extension-free imports.

https://webpack.js.org/configuration/resolve/#resolve-extensions

Comment (khumphrey) [reply app/main.jsx]: Good points! Fixed 😄

Comment (khumphrey) [reply app/main.jsx]: Any other thoughts before I merge?

Comment (collin) [reply app/main.jsx]: Ah, looks like we already have the resolver set up. No other thoughts here.

**Pull request merged**